app/services/archive/undeclared_profile.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Archive
+  class UndeclaredProfile < UnvalidatedProfile
+  private
+
+    attr_accessor :cohort
+
+    def initialize(participant_profile, cohort_year: 2021, reason: "undeclared participants in 2021", keep_original: false)
+      super(participant_profile, reason:, keep_original:)
+
+      @cohort = Cohort.find_by(start_year: cohort_year)
+    end
+
+    def check_profile_can_be_archived!
+      if profile_not_in_requested_cohort?
+        raise ArchiveError, "Profile #{participant_profile.id} is not in #{cohort.start_year} cohort"
+      elsif profile_has_declarations?
```

Comment:
The method `ParticipantProfile::ECF#archivable?` will do all the checks for you

Reply:
That is worrying because there is already another flavour of archiving that this derives from where the checks are slightly different

Reply:
Agreed.